### PR TITLE
Some builders

### PR DESCRIPTION
**Pull request by rubendm92**
### Summary

I've implemented builders for effect state.set and coeffects state.get and global.get. I haven't done anymore because I wanted feedback if this the way we'll like to expose the builders.

You can see some usages in the react todos example.

### CODE REVIEW

File: examples/with-react-todos/src/todos/TodoList/events.js [hunk 1]
```diff
@@ -28,18 +29,14 @@ export default function registerTodoListEvents() {
 
     const todos = extractTodos(response);
 
-    return {
-      'state.set': { todos: todos },
-    };
+    return stateBuilder.set({ todos: todos });
```
Comment (trikitrok) [inline examples/with-react-todos/src/todos/TodoList/events.js:33]: I'd prefer to read `state.set({ todos: todos });`

File: examples/with-react-todos/src/todos/TodoList/events.js [hunk 2]
```diff
@@ -12,7 +13,7 @@ export default function registerTodoListEvents() {
         },
       };
     },
-    [coeffect('global.get', 'apiUrl')]
+    [globalBuilder.get('apiUrl')]
```
Comment (trikitrok) [inline examples/with-react-todos/src/todos/TodoList/events.js:16]: I'd prefer to read `global.get('apiUrl')`

File: examples/with-react-todos/src/todos/TodoList/events.js [hunk 3]
```diff
@@ -61,15 +58,15 @@ export default function registerTodoListEvents() {
...
-        },
+        }),
         toast: {
           text: `"${text}" was marked as ${isDone ? 'undone' : 'done'}.`,
           milliseconds: 3000,
         },
       };
     },
-    [coeffect('state.get', { todos: 'todos' })]
+    [stateBuilder.get({ todos: 'todos' })]
```
Comment (trikitrok) [inline examples/with-react-todos/src/todos/TodoList/events.js:71]: I'd prefer to read `state.get({ todos: 'todos' })`

File: examples/with-react-todos/src/todos/TodoList/events.js (hunk 2 above)
Comment (rubendm92) [reply examples/with-react-todos/src/todos/TodoList/events.js:16]: Problem here is that `global` is already a declared variable in Node, so I can not call it like that :(

File: examples/with-react-todos/src/todos/TodoList/events.js (hunk 3 above)
Comment (trikitrok) [reply examples/with-react-todos/src/todos/TodoList/events.js:71]: That's true :(

**Comment by rubendm92** [comment]
> It's really great.
> 
> I added some comments regarding the naming (it happens in more places thatn the ones I commented).
> 
> The idea behind those comments is that I'd rather keep using the coeffects & effects names directly instead of coeffectBuilder & effectBuilder because in my opinion it's shorter and reads better (also looking at the way these ones are used, they seem more like factories than builders).
> 
> Example:
> Using `state.set({ todos: todos });` instead of `stateBuilder.set({ todos: todos });`
> 
> Another option might be using a factory `stateSet({ todos: todos });` which I think is enough for these simple coeffects & effects.
> 
> I think the need for builders will come when creating more complicated coeffects or effects, such as the http effects that have many parameters of which some are optional.

Thanks for the feedback!

I also prefer the `state.set({ todos: todos })` format, so I changed it to that. I had to name the `global.get` coeffect as `globalVariable.get` since I can not use `global` as a name. If you find a better name I'm open to suggestions :)

**Comment by trikitrok** [comment]
`globalVariable.get` reads great!!

**Pull request closed without merging**